motoko/internet_identity_integration/src/greet_frontend/src/index.js

```diff
@@ -1,48 +1,61 @@
-import {createActor, greet_backend} from "../../declarations/greet_backend";
-import {AuthClient} from "@dfinity/auth-client"
-import {HttpAgent} from "@dfinity/agent";
+import { createActor, greet_backend } from "../../declarations/greet_backend";
+import { AuthClient } from "@dfinity/auth-client";
+import { HttpAgent } from "@dfinity/agent";
 
-let actor = greet_backend;
+// Global variables that we will set up
+window.greetActor;
+window.authClient;
 
 const greetButton = document.getElementById("greet");
-greetButton.onclick = async (e) => {
-    e.preventDefault();
+const loginButton = document.getElementById("login");
 
-    greetButton.setAttribute("disabled", true);
+AuthClient.create().then((client) => {
+  // Put this on the window so we can access it from the console. This is helpful for learning, but it's recommended to keep it in a closure in a real app.
+  window.authClient = client;
+  // We can use the authClient to check if the user is already authenticated
+  setActor(client);
+
+  // Now that the auth client is created, we can enable the login and greet buttons
+  loginButton.removeAttribute("disabled");
+  greetButton.removeAttribute("disabled");
+});
+
+loginButton.onclick = () => {
+  // start the login process and wait for it to finish
+  window.authClient.login({
+    identityProvider: process.env.II_URL,
+    onSuccess: () => {
+      setActor(window.authClient);
```

Comment:
Do you remove the login button onSuccess?

Reply:
No, I didn't notice that logic in the original version